textselect.cpp

```diff
@@ -150,11 +183,53 @@ void TextSelect::handleMouseDown(const ImVec2& cursorPosStart) {
     const float textHeight = ImGui::GetTextLineHeightWithSpacing();
     ImVec2 mousePos = ImGui::GetMousePos() - cursorPosStart;
 
-    // Get Y position of mouse cursor, in terms of line number (clamped to the valid range)
-    std::size_t y = static_cast<std::size_t>(std::min(std::max(std::floor(mousePos.y / textHeight), 0.0f), static_cast<float>(numLines - 1)));
+    std::string_view currentLine;
+    std::size_t x, y;
+    if (enableWordWrap) {
+        ImGuiWindow *window = ImGui::GetCurrentWindow();
+
+        const float wrapWidth = ImGui::CalcWrapWidthForPos(window->DC.CursorPos, 0);
+
+        // Find the index of the line under the cursor and its size.
+        float accumulatedHeight = 0;
+        ImVec2 currentLineSize = {};
+        y = numLines - 1;
+        for (std::size_t i = 0; i < numLines - 1; ++i) {
+            std::string_view line = getLineAtIdx(i);
+            currentLineSize = ImGui::CalcTextSize(line.data(), line.data() + line.size(), false, wrapWidth);
+
+            if (mousePos.y < accumulatedHeight + currentLineSize.y) {
+                y = i;
+                break;
+            }
+
+            accumulatedHeight += currentLineSize.y + ImGui::GetCurrentContext()->Style.ItemSpacing.y;
+        }
+        if (y >= numLines)
+            return;
```

Comment:
Please ensure the bodies of single-line `if` statements are surrounded by braces.

```suggestion
        if (y >= numLines) {
            return;
        }
```